examples/main/main.cpp

```diff
@@ -118,6 +120,16 @@ static void llama_log_callback_logTee(ggml_log_level level, const char * text, v
 }
 
 int main(int argc, char ** argv) {
+#ifndef _MSC_VER
+    // Check if we have an external attachment to a file descriptor for out of band control tokens (e.g. bash `3>/dev/null` )
+    // Placed here to avoid file descriptor being polluted by gpt_params_parse() opening files
+    const bool control_token_file_descriptor_is_attached = fcntl(SPECIAL_FILENO, F_GETFL) != -1;
+    if (!control_token_file_descriptor_is_attached) {
+        // Duplicate stdout file descriptor to control token file descriptor to merge the two streams
+        dup2(STDOUT_FILENO, SPECIAL_FILENO);
+    }
+#endif
```

Comment:
I don't think this is needed. If one wants to debug special/control tokens, then they don't pass `--no-special`. For script usage where special tokens are not wanted, you pass `--no-special`. The extra file descriptor here is too complicated

Reply:
Jart has merged in the adjustment to this PR now.

Will wait a bit further for any more feedback, but I think we have now reached a consensus point for this PR and will merge this in a day or two if no other problems emerges (Or immediate upon positive confirmation from you).